Support configuring the logger with `options.service`, etc.

rmharrison opened the pull request:
If this PR fixes a bug, you _must_ add test cases representative of the bug.

#### What does this PR do?
Previously `options.service` test was commented because passing options to the logger wasn't yet supported.
```
# test/format.test.js
// expect(parsedTestMess.service).toEqual(options.service);
```

#### Related JIRA tickets:

#### How should this be manually tested?
Follow README.md instructions and see that the options output to the logger
```
> logger.info('message', {foo: 'bar', baz: 'qux'});
{"service":"test-service","logger":"Winston-JSON-Formatter","hostname":"<>","level":"info","msg":"message","meta":{"service":{"version":"1.0.0","node_env":""},"logger":{"time":"2018-11-28T02:52:06.700Z"},"event":{"foo":"bar","baz":"qux"}},"err":{}}
```

#### Background/Context

#### Screenshots (if appropriate):

rmharrison committed:
Support configuring the logger with `options.service`, `options.version`, etc.

morleegan commented on test/format.test.js:
```diff
@@ -14,86 +14,100 @@ const options = {
         beforeEach(() => {
-            // spy to check output
             spy = jest.fn((info) => {
+                expect(spy).toHaveBeenCalled();
```
`testLoggerInfo` is also calling this check. Remove from here and leave in `testLoggerInfo`

Commit:
updated gitignore, removed line

morleegan reviewed (approved):
LGTM

morleegan commented:
@rmharrison - leaving for you to merge

rmharrison commented:
Holding-off on merging pending getting CI setup.

rmharrison committed:
Update README example to use `level: silly`

- Previously, log levels below `level: info `(default) were not output to Console
- Test coverage (statement) now 100%
- Added CHANGELOG.md

rmharrison committed:
Add `.env.test`

rmharrison committed:
Reconfigure package for npm

- Move `index.js` to src
- Don't use ES6 (no babel required) in README

rmharrison commented:
"Do not merge" removed because Jenkins CI is now setup.
Thanks @nickmccally

rmharrison commented:
@puckybreg : Let me know when you've had a look, then I'll merge and cut release `0.9.0` (manual `npm publish` for the time being).

puckybreg reviewed (approved):
👍
